packages/origin-247-certificate/src/offchain-certificate/synchronize/strategies/batch-synchronize.strategy.ts

```diff
@@ -0,0 +1,37 @@
+import { SynchronizeStrategy } from './synchronize.strategy';
+import {
+    CertificateEventRepository,
+    SynchronizableEvent
+} from '../../repositories/CertificateEvent/CertificateEvent.repository';
+import { Inject, Injectable } from '@nestjs/common';
+import { CertificateCommandRepository } from '../../repositories/CertificateCommand/CertificateCommand.repository';
+import {
+    CERTIFICATE_COMMAND_REPOSITORY,
+    CERTIFICATE_EVENT_REPOSITORY
+} from '../../repositories/repository.keys';
+import { PersistProcessor } from '../handlers/persist.handler';
+import { CertificateEventType } from '../../events/Certificate.events';
+
+@Injectable()
+export class BatchSynchronizeStrategy implements SynchronizeStrategy {
```

Comment:
do you handle the fact, that if previous event for given certificate failed, you should not try to persist any following events for this certificate?

Reply:
fixed